### ISSUE TITLE
MappedByteBuffer is not a valid flatbuffer model TFLITE Model_conversion_issue

### ISSUE THREAD

**Issue by Raj-08** [tensorflow/tensorflow#22584]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: 
- **TensorFlow version (use command below)**: 1.10
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.15.2
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 9.0
- **GPU model and memory**: NVIDIA 12 GB
- **Exact command to reproduce**:

I have a model in which i have connected two mobilenet models . These are the following steps:

1. Train the model. - Success
2. Freeze the graph using eval graph - Success
///Defiine the graph then 
` 
  with tf.Session() as sess:

        saver.restore(sess, '/home/ubuntu/raj/research/trimap_mobnet/model.ckpt-205740')
        g = tf.get_default_graph()
        tf.contrib.quantize.create_eval_graph(input_graph=g)
        sess.run(tf.global_variables_initializer())
        saver.restore(sess, '/home/ubuntu/raj/research/trimap_mobnet/model.ckpt-205740')
        input_graph_def = graph.as_graph_def()
        output_graph_def = graph_util.convert_variables_to_constants(
            sess,
            input_graph_def,
           [_OUTPUT_NAME])        
        with tf.gfile.GFile(output_graph_name, "wb") as f:
            f.write(output_graph_def.SerializeToString())`
3. Toco convert -Success

tflite_convert  --output_file=./lite.tflite  --input=./frozen.pb --input_arrays=image_ph --output_arrays=SemanticPredictions --input_shapes=1,481,481,3

4 . Load the model on phone - Error

While loading the lite model on phone by running the command

`var localSource = FirebaseLocalModelSource.Builder("my_local_model")
               .setAssetFilePath("lite.tflite")
               .build()
       FirebaseModelManager.getInstance().registerLocalModelSource(localSource)
        var options = FirebaseModelOptions.Builder()
               .setLocalModelName("my_local_model")
               .build()
       firebaseInterpreter = FirebaseModelInterpreter.getInstance(options)
       inputOutputOptions = FirebaseModelInputOutputOptions.Builder()
               .setInputFormat(0, FirebaseModelDataType.FLOAT32, intArrayOf(1, 481, 481, 3))
               .setOutputFormat(0, FirebaseModelDataType.FLOAT32, intArrayOf(1, 481, 481, 3))
               .build()`

 i get the following error:
MappedByteBuffer is not a valid flatbuffer model

Any reason why am i getting this error or is it a tensorflow converter bug?`

**Comment by Raj-08** [comment]
In detail here is what i am doing:

IMAGE >> MOBILENET MODEL1 >> OUTPUT1 >> SCALAR MULTIPLICATION OF OUTPUT1 with 255 >> CONCAT THIS WITH IMAGE >>MOBILENET MODEL2 >>OUTPUT2>>SCALAR MULTIPLICATION OF OUTPUT2 WITH 255 >> CONCAT THIS WITH IMAGE>> 3 LAYERS OF 3*3 CONVOLUTIONS>>  SIGMOID >> PREDICTION >> SPLIT IMAGE IN 3 CHANNELS >> MULTIPLY PREDICTION WITH ALL 3 CHANNELS >> CONCAT 3 CHANNELS  >> FINAL OUTPUT

**Comment by Raj-08** [comment]
After removing Deconvolution and replacing it with bilinear resizing , i get the following error : 

Check failed : input_array_shape.dimension_count()==output_array_shape.dimension_count() (5 vs 4)

**Comment by Raj-08** [comment]
Any update on this ??

**Comment by wt-huang** [comment]
@Raj-08 You would need to check whether there are some ops in your model not supported by tflite. To replace Deconvolution, you can apply `tf.nn.conv2d_transpose` rather than bilinear resizing. Also make sure the shape of input and output array match. Lastly, use the latest TensorFlow version.

**Comment by Raj-08** [comment]
Sure will try that out !!

**Comment by Raj-08** [comment]
I used tf.nn.conv2d_transpose everywhere replacing it with bilinear resizing. 
Still i get the same error  MappedByteBuffer is not a valid flatbuffer model

**Comment by wt-huang** [comment]
I was able to convert a simple model with `tf.nn.conv2d_transpose` without running errors.  You may have other ops in your model that are not supported by tflite.

**Comment by wt-huang** [comment]
@Raj-08 Yes, argmax and argmin are supported in TensorFlow 1.11.0

**Comment by wt-huang** [comment]
Closing this, feel free to reopen if issue persists.

**Comment by zhang-jian** [comment]
> I was able to convert a simple model with `tf.nn.conv2d_transpose` without running errors. You may have other ops in your model that are not supported by tflite.

Sorry to post on the closed issue.
@wt-huang  did you test your tf.nn.conv2d_transpose outputs, i couldn't get expected results as in https://github.com/tensorflow/tensorflow/issues/25462, thanks.
